### PR TITLE
Dump Pod Status in logs instead of complete PodSpec

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview
As of now we used to dump all Pod Spec in logs in case of error or pod failure. This PR is to dump only the Pod Status and not all Spec.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #1464

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

Controller logs that shows only PodStatus for a Failed Mysql blueprint actionSet [https://gist.github.com/akankshakumari393/3402379b3e6440f5ecd934af5204b649](https://gist.github.com/akankshakumari393/3402379b3e6440f5ecd934af5204b649)

### CODE REVIEW

File: pkg/kube/pod.go [hunk 1]
```diff
@@ -299,7 +299,7 @@ func WaitForPodCompletion(ctx context.Context, cli kubernetes.Interface, namespa
 		}
 		switch p.Status.Phase {
 		case v1.PodFailed:
-			return false, errors.Errorf("Pod %s failed. Pod details (%s)", name, log.SafeDumpPodObject(p))
+			return false, errors.Errorf("Pod %s failed. Pod Status details (%s)", name, log.SafeDumpPodStatusObject(p))
```
Comment (pavannd1) [inline pkg/kube/pod.go:302]: We can just use `p.Status.String()` directly. Changes to helper and the tests will not be necessary

**Review by ihcsim — APPROVED** [review]
LGTM - barring https://github.com/kanisterio/kanister/pull/1506/files#r906314627.

File: pkg/kube/pod.go (hunk 1 above)
Comment (ihcsim) [reply pkg/kube/pod.go:302]: Agreed - then we can also remove `log.SafeDumpPodObject()`.

File: pkg/kube/pod.go [hunk 2]
```diff
@@ -299,7 +299,7 @@ func WaitForPodCompletion(ctx context.Context, cli kubernetes.Interface, namespa
 		}
 		switch p.Status.Phase {
 		case v1.PodFailed:
-			return false, errors.Errorf("Pod %s failed. Pod details (%s)", name, log.SafeDumpPodObject(p))
+			return false, errors.Errorf("Pod %s failed. Pod Status details (%s)", name, p.Status.String())
```
Comment (pavannd1) [inline pkg/kube/pod.go:302]:
```suggestion
			return false, errors.Errorf("Pod %s failed. Pod status: %s", name, p.Status.String())
```

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**